rust/cubesql/cubesql/src/compile/rewrite/rules/members.rs

```diff
@@ -1478,6 +1511,78 @@ impl MemberRules {
             )
     }
 
+    fn select_distinct_dimensions(
+        &self,
+        alias_to_cube_var: &'static str,
+        members_var: &'static str,
+        filters_var: &'static str,
+        left_ungrouped_var: &'static str,
+    ) -> impl Fn(&mut CubeEGraph, &mut Subst) -> bool {
+        let alias_to_cube_var = var!(alias_to_cube_var);
+        let members_var = var!(members_var);
+        let filters_var = var!(filters_var);
+        let left_ungrouped_var = var!(left_ungrouped_var);
+        let meta_context = self.meta_context.clone();
+
+        move |egraph, subst| {
+            let empty_filters = &egraph
+                .index(subst[filters_var])
```

Comment:
You can use `&egraph[subst[filters_var]]` instead of ` &egraph.index(subst[filters_var])`. IMO it looks cleaner.